apps/web/ee/lib/impersonation/ImpersonationProvider.ts

```diff
@@ -0,0 +1,62 @@
+import CredentialsProvider from "next-auth/providers/credentials";
+import { getSession } from "next-auth/react";
+
+import prisma from "@lib/prisma";
+
+const ImpersonationProvider = CredentialsProvider({
+  id: "impersonation-auth",
+  name: "Impersonation",
+  type: "credentials",
+  credentials: {
+    username: { label: "Username", type: "text " },
+  },
+  async authorize(creds, req) {
+    // @ts-ignore need to figure out how to correctly type this
+    const session = await getSession({ req });
+    if (session?.user.role != "ADMIN" || !session) {
+      throw new Error("You do not have permission to do this.");
+    }
+
```

Comment:
We should maybe query the user from the session with prisma to ensure it is admin but im sure this is reliable enough as its ran on the server?

Reply:
I think this should be safe enough, no need to add an additional database call since session comes from database data anyways? Maybe since it's an internal feature, it wouldn't hurt though, since we actually still use JWT for sessions I guess if someone where to stole the ENV_SECRET for generating those could theoretically generate sessions with ADMIN role on it? @zomars @emrysal @baileypumfleet thoughts?

Reply:
If someone stole the `ENV_SECRET` we should be in way more bigger trouble than that.